src/PrestaShopBundle/Form/Admin/Type/NavigationTabType.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Type;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * This form type is used as a container of sub forms, each sub form will be rendered as a part of navigation tab
+ * component. Each first level child is used as a different tab, its label is used for the tab name and it's widget
+ * as the tab content.
+ */
+class NavigationTabType extends AbstractType
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        parent::buildForm($builder, $options);
+        if (!empty($options['toolbar_buttons'])) {
+            if ($builder->has('toolbar_buttons')) {
+                $this->logger->warning('You should not add a field which name is toolbar_buttons on this component as it is used internally.');
```

Comment:
```suggestion
                $this->logger->warning('You should not add an option "toolbar_buttons" in this component as such option is used internally.');
```

Reply:
No, actually I'm not talking about an option here, but really a form field.

In the following lines, I'm doing `$builder->add('toolbar_buttons', ...)` meaning the `form.tollbar_buttons` is now a new field which type is `ButtonCollectionType` I need it to render it in the twig theme thus relying on Form component and twig tools.

But the field name is hard-coded here, meaning if a developer already used it then it will be overridden, that's why I check if the field already existed. I hesitated with an exception but I thought maybe it's a bit brutal, but maybe it would actually be easier to understand for the developer. Thus a clear exception is breaking everything instead of a field that magically disappears.

An alternative could be to throw an exception in debug mode and log a warning in prod mode. What do you think?

Reply:
that sounds like a good alternative 👍 

also as we discussed, maybe adding an underscore like this `_ toolbar_buttons` would reduce the risk of naming overlap